open_bounties/0119 Frontend POAP Web APP.md

```diff
@@ -32,33 +31,126 @@ Provide event organizers with a no-code tool for distributing NFTs for attendanc
+
+Ultimately, there are three types of users:
+1. Event organizers
+2. Event attendees
+3. The Admin / Tool Operator (needs technical expertise)
+
+### Event organizers need to be able to
+1. Create an event
+2. Cancel an event
+3. Get a link to share with attendees for attendees to redeem the NFT
+4. See how many NFTs are minted / claimed
+5. See who has claimed the NFTs
+   - They also need to be able to search for an account via their wallet address
+
+### Attendees need to be able to
+1. Claim an NFT if they have the event link
+2. Verify ownership of the NFT (to prove they’re the account holder)
+
+### The Admin needs to be able to:
+1. View all events on their platform
+   - But shouldn’t have access to the links to redeem NFTs
+2. Cancel any event on their platform
+   - In order to free up their reserve requirements
+
+**In order to implement these action items we expect to need several pages:**
+
+An “Organizer Home” page where event organizers can log in and manage their events.
+An “Organizer Event Details Page” which provides additional info on an event.
+An “Attendee Event” page where attendees can go to claim an NFT (optional idea: this could be linked via QR code during the event as a way to distribute the link)
+This would also have event-specific information if the organizer is logged in with their wallet.
+An “Admin” page for the 3rd party operator (“Admin”) to configure things
+
+### Organizer Home Page before Login
+
+1. A clean and straightforward interface that welcomes users to the platform.
+2. Button for "Organizer Login" Leading to XUMM wallet connection.
+
+### Organizer Home Page after Login:
+1. A welcoming personalized greeting, including the Organizer's account address.
+2. Overview of the Organizer's created events: 
+   - This could be a list or cards displaying essential information about each event (name, date, number of attendees, status).
+   - The cards should indicate the status of the event
+      1. Ongoing
+      2. Completed
+      3. Canceled
+3. A way to “Create a New Event”
+4. The "Create New Event" option should lead to a detailed form where the organizer can specify event details 
+   - Start Date
+   - End Date
+   - Description
+   - Graphic
+   - Event ID
+      - Which must be unique across all events to allow for looking up the event after the fact
+   - The number of NFTs to create (Capped to 200 attendees)
+5. At the bottom of the "Create New Event" form, there should be a live preview of the reserve required (preferably with UI showing the formula being used to calculate + a link to the docs to avoid confusions around why the reserve is required).
```

Comment:
This should include more than just the reserve requirement but also fees needed for minting and creating offers.

Reply:
I was including that in the term "reserve" since the primary cost there is the incremental reserve (as opposed to the account reserve). I think the reserve calculations at the bottom of the requirements factors in both the NFT reserve and the reserve for creating offers to sell them.

I'll add +1 XRP for the transaction fees though, I forgot that that'd be slightly above the actual reserve requirements.

Reply:
I think it makes sense to also include something to cover transaction costs. However, I would suggest to keep the remainder of that 1 XRP deposit as a sort of "event creation charge". This could help to avoid abuse of someone constantly creating and canceling events. Possibly also cover some of the operational costs.

Reply:
I think it’s important the POAP App be free to use since we’re sending a large amount of XRP specifically for the reserve, and it could be confusing if the amount of XRP that comes back is anything different. It could create worry that there are hidden fees for different operations on the website. 

Because the computation that gets triggered from minting / canceling are pretty simple computationally (even if they take some time because they’ve got to talk to the ledger) so I’m not too worried about the abuse of a manual user clicking around on the UI. (If they’re botting, they can probably find other ways to break the website that we don’t care to deal with in this bounty).